### DISCUSSION TITLE
Let's get started!

### DISCUSSION THREAD

**Discussion by nattyg93** [vim-vix-remix/vim-vix-remix#4] [General]
Hi folks,

After creating this organisation things became very busy for me at work, which was then followed by the silly season, so I have not had any free time before now.

I would like to thank everyone that has volunteered to work on the project and to the people that have made a start on getting things under-way.

To begin with, there has been a bit of discussion here and there about where we should discuss things and some people have suggested matrix. This is an open source project with a strong community behind it, so I would like to keep the communication as open and transparent as possible, and am in favour of keeping as much of the discussion publicly visible as possible. So I am personally happy to handle most of the communication here on Github through discussions and issues. I am not suggesting we cannot do this on matrix, and I am open to being convinced otherwise if anyone wants to make the case for a particular platform that would suit the community and maintainers best.

To get the ball rolling, we need to ensure that we are on the same page about what our goals are. It would be great to get some input from the other maintainers (@victorz, @codeDude64, @LordPax) and the community at large too.

As we have discussed, we want to add this extension to the Firefox Add-on store, so we need to identify what steps we need to take. The broad outline of the steps required are on the [comment here](https://github.com/ueokande/vim-vixen/pull/1437#issuecomment-1278659206). Is there any chance you could give us some more detail about what is required, @jman-schief?

Of course, working out CI/CD will be critical as we move forward, but I think at least initially, we should aim to get a working build submitted first and foremost.

As mentioned in the comment I have linked above we will need a verified AMO account, which I am willing to be responsible - I will try to get that set up within the next week.

Please feel free to contribute any other points of discussion that I have missed out. Again, I am new to contributing to open source, so I apologise for any missteps as we figure all of this out.

Cheers,
Nat

**Comment by toastal** [comment]
What is the Matrix room? And is it bridged to Libera Chat?

**Reply by nattyg93** [reply]
Hi @toastal, I'll quote from the message above regarding the use of Matrix.

> To begin with, there has been a bit of discussion here and there about where we should discuss things and some people have suggested matrix. This is an open source project with a strong community behind it, so I would like to keep the communication as open and transparent as possible, and am in favour of keeping as much of the discussion publicly visible as possible. So I am personally happy to handle most of the communication here on Github through discussions and issues. I am not suggesting we cannot do this on matrix, and I am open to being convinced otherwise if anyone wants to make the case for a particular platform that would suit the community and maintainers best.

**Reply by jiriks74** [reply]
There is none so far, as far as I know.

I think that it would be a nice idea because there could be more "casual" conversation over there and if we come up with something we can add it in here, so it stays clear of any clutter where people submit 100 issues discussing one thing. It's not a problem IMO to just add a matrix invite to the README so that people can join. I would be willing to have Element opened on my PC overlooking the room so that some rando cannot wreak havoc. I wouldn't be doing it 24/7, of course, but at least someone would be looking over it and it shouldn't be really hard as there wouldn't be much activity, at least from the start.

**Reply by toastal** [reply]
I believe it's like a single-click flow to get bridging to Libera Chat. This helps accessibility for those on devices and connections that can't handle Matrix. This said, the room still isn't linked or mentioned. (Also, I appreciate that Discord/Slack were _not_ mentioned).

**Reply by jiriks74** [reply]
> I believe it's like a single-click flow to get bridging to Libera Chat. This helps accessibility for those on devices and connections that can't handle Matrix. This said, the room still isn't linked or mentioned. (Also, I appreciate that Discord/Slack were _not_ mentioned).

I'll look into creating the room tomorrow evening if others are interested. I'll post it here and I'd modify the README with the new branding and the matrix room. Will you add me or should I make a pull request?

**Comment by jiriks74** [comment]
The goals?

Don't hate on me, but I can so far thing of only these basic things (if you wouldn't mind we could track them here for now until we find a better way):

- [ ] Change the branding from `vim-vixen` to `vim-vix-remix`
- [ ] Get the patched extension to the store so that people can start using it
  - [ ] AMO account
  - [ ] Working build
  - [ ] Published working build
- [ ] Familiarize ourselves with the base
  - [ ] [Figure out what we have to do to port this to Manifest V3](https://github.com/ueokande/vim-vixen/blob/62f4610f2b9e3d49ac4b3f09074f543939720ffb/manifest.json#L2)
- [ ] Figure out CI/CD
  - Firefox, as far as I know, plans to discontinue it. Not right now, but once they implement their extra features to Manifest V3 they will.
- [ ] Transfer some issues we think would be nice to fix/implement from the original repo
  - [ ] The same goes for PRs

Please tell me what to add/change/remove

**Reply by victorz** [reply]
By "Firefox plans to discontinue 'it'", by "it", you mean Continuous Delivery to the Add-ons "store"? Or something else?

**Reply by jman-schief** [reply]
No established deadline to discontinue manifest V2 (perhaps relevant [Mozilla blog post](https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/))

**Comment by codeDude64** [comment]
Should we considerate use a Kanban/scrum tools?
I don't mean use a tight scrum but I think it's a good idea create a Kanban board, Usually I use Jira in my job but I can suggest Trello I used it in the college. Even GitHub has its own Kanban board where we are allow to link the issues and PR.
It allow us to handle who is working in something witch tickets or task are necessary (priority) etc.

**Reply by jiriks74** [reply]
I think that using the GitHub one would be enough, or even  better in some cases. But I wouldn't push it as of now. IMO we should first do the basics: rebrand the extension and publish it. Then I think we could look into the codebase and learn it. I have already a feature that I think would be cool to implement, disabling autofocus for text fields, but IDK if it's even possible and there's no point in looking into it if we aren't able to deliver the extension.

But yes, I think that as we manage to get this movin a bit I think that we should find some tools to help us organize.

**Comment by jman-schief** [comment]
> As we have discussed, we want to add this extension to the Firefox Add-on store, so we need to identify what steps we need to take. The broad outline of the steps required are on the [comment here](https://github.com/ueokande/vim-vixen/pull/1437#issuecomment-1278659206). Is there any chance you could give us some more detail about what is required, @jman-schief?

While you guys figure out a place for discussions, I'll reply here and expand a little bit on my previous comment.

So, the process of publishing a new extension on AMO is rather straightforward. Someone taking ownership of the publishing should create an AMO account (which I understand could be you @nattyg93 :tada:). I would enable from the start 2FA authentication on this AMO account for additional security against credentials leak/thief. The standard way to publish extensions on AMO is using the [Mozilla web-ext tool](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#web-ext_sign), the instructions to sign and publish the build artifact should be pretty straightforward either.

I would instead look first at reproducing a working build of Vim Vixen, see the `CONTRIBUTING.md`. To package the extension I'd look at how it's done in the [GitHub build workflow](https://github.com/ueokande/vim-vixen/actions/runs/4049996133/workflow#L7).

Speaking of Vim Vixen [CI latest builds](https://github.com/ueokande/vim-vixen/actions), I would expect some bumps on the road when building the package, I see some failures when upgrading dependencies. You can replicate in this repository the build GitHub Action from upstream.
**Note**: I would absolutely remove the "Release to AMO" step from that CI workflow and instead use the above mentioned tool `web-ext` to publish, I've read too many horror stories of automated publishing of software packages from CIs artifacts infected by malware.

Once it builds and "works on your machine" (:tm: 2023), it's time to tweak the `manifest.json` with a new name and version, you want people to identify at a glance which one to install when seeing two Vim Vixen extensions on the AMO store. Why should I install Vim Vixen Remix? Maybe a variation of the icon would also help?

**Comment by carsonhenrich** [comment]
Hey guys, I just happened upon the discussion in the original vim-vixen repo and thought I might be able to with this project. I don't have any experience working on an open-source project so I don't know how much I'll be able to assist but I'd like to start contributing to more projects so I will help however I am able.

**Reply by codeDude64** [reply]
It's my first open-source project as well. I think it's fine to have more hands involved in this. For me it's ok

**Comment by schr4der** [comment]
Hey all, I just read-up on the past discussion and would like to be involved. As some of you, I'm also new to open source development, so it would be nice to start with a project I use.

**Comment by bovine3dom** [comment]
Hi, I develop Tridactyl, a similar extension to Vim Vixen.

I would recommend that you try very hard _not_ to fork Vim Vixen; it would be best for users if they just automatically got updates since you don't seem to have wildly different opinions to ueokande about the direction of the project.

I have just sent ueokande an email asking if he would be OK letting me make Vixen releases. I'll let you know if I get a reply.


In the meantime you could open a PR from this fork's master branch on Vim Vixen's repository to increase visibility of the fork.

**Reply by victorz** [reply]
> I would recommend that you try very hard not to fork Vim Vixen

I think most of us did :slightly_smiling_face:, but I agree with the sentiment, definitely.

This would be an excellent solution in my opinion. Let's hope he replies!

**Comment by ueokande** [comment]
Hi!  I'm Vim Vixen's author.  
Thanks for informing me about this project, and I'm honored to reach this discussion.

As you are already aware, the Vim Vixen is no longer actively developed now.  I cannot release new versions of vim vixen onto both GitHub and AMO anymore.  But that is not the intent I close the project.  I have been working to develop a forked one with a new repository and new branding, [vimmatic](https://github.com/ueokande/vimmatic).  This project has some features; Manifest v3, Chromium support, and renewal architecture. 

I am preparing an experimental release of the vimmatic project.  Also, I will announce the project's status of the vim-vixen on GitHub and make the repository archived before long.  Of course, I welcome you to start your project under the open-source license.

Thank you very much.

**Reply by bovine3dom** [reply]
That's great news, I can't wait to see what you do with vimmatic!

> I cannot release new versions of vim vixen onto both GitHub and AMO anymore

If it's a problem with the AMO (e.g. you have lost access to your account) can I suggest emailing `amo-admins@mozilla.com` and seeing if they can help? They're friendly :)

**Reply by victorz** [reply]
Looking forward to seeing vimmatic for Firefox if it ever happens! Will definitely keep an eye on it.

Would you please update the README on the vim-vixen repo, or archive it, so it's clear that it is abandoned? Please and thank you! :pray:

**Reply by AlexeySachkov** [reply]
Hi @ueokande, it is great to see that the project is being reincarnated, but have you though about maybe creating a GitHub organization to make the project community-owned (or community-co-owned)? I'm sure that you should be able to find contributors whom you trust enough to make them co-owners so if you don't have time (for example), the project still moves forward.

No offence, but for an arbitrary user or contributor there is still no guarantee that the new project won't go silent as `vim-vixen` at some point. Having more than one person behind the project who is able to merge changes and do releases greatly increases reliability and lifespan of the project.

**Comment by nattyg93** [comment]
Hi folks,

It's now been over a month since I first posted and I've not been able to find any time to contribute to this project.

Between the lack of time for this project and in light of the comments from @ueokande and @bovine3dom above, I've decided to ~~not~~ step away from the project. This doesn't mean the project can't continue forward, but I need to be realistic about the amount of time and effort I can contribute and at this point it is not enough.

Hopefully the rest of the maintainers can find time to move forward with this fantastic project.

Cheers,
Nat

**Reply by victorz** [reply]
> I've decided to not step away from the project

*\*now* step away from the project, I'm assuming?

**Comment by toastal** [comment]
At that rate, why not move to Codeberg or elsewhere so the project & contributors are not locked to a closed-source code forge
